storm-server/src/main/java/org/apache/storm/scheduler/blacklist/BlacklistScheduler.java

```diff
@@ -155,6 +163,18 @@ private void badSupervisors(Map<String, SupervisorDetails> supervisors) {
         badSupervisorsToleranceSlidingWindow.add(badSupervisors);
     }
 
+    private void trackAssignmentFailures() {
+        if (!blacklistSendAssignentFailures) {
+            return;
+        }
+        Map<String, Integer> assignmentFailureWindow = new HashMap<>();
+        assignmentFailureWindow.putAll(this.assignmentFailures);
+        synchronized (assignmentFailures) {
```

Comment:
Since we already `synchronized` on it, maybe should just as well put `assignmentFailureWindow.putAll(this.assignmentFailures);` inside the block and then it doesn't need to be a `concurrentHashMap`.  
To reduce two synchronization mechanisms to one

Reply:
yes